### PR TITLE
KEYCLOAK-17781 Make exists and notExists collection aware

### PR DESCRIPTION
**Pull request by mhajas**

### CODE REVIEW

File: model/map/src/main/java/org/keycloak/models/map/storage/CriteriaOperator.java [hunk 1]
```diff
@@ -106,14 +106,38 @@ private static Object getFirstArrayElement(Object[] value) throws IllegalStateEx
...
             throw new IllegalStateException("Invalid argument: " + Arrays.toString(value));
         }
-        return Objects::nonNull;
+        
+        return CriteriaOperator::collectionAwareExists;
+    }
+    
+    private static boolean collectionAwareExists(Object checkedObject) {
+        if (Objects.nonNull(checkedObject)) {
+            if (checkedObject instanceof Collection) {
```
Comment (hmlnarik) [inline model/map/src/main/java/org/keycloak/models/map/storage/CriteriaOperator.java:115]: This is unnecessary, `instanceof` is `false` if `checkedObject` is `null`

**Review by hmlnarik — COMMENTED** [review]
Thanks, looks good to me! Please see one small nit inline

File: model/map/src/main/java/org/keycloak/models/map/storage/CriteriaOperator.java (hunk 1 above)
Comment (mhajas) [reply model/map/src/main/java/org/keycloak/models/map/storage/CriteriaOperator.java]: Thanks. Updated.

**Comment by mhajas** [comment]
@hmlnarik I updated the PR with suggested change and opened it for reviews. Maybe @vramik would like to have a look as it touches Realm map store.

**Review by vramik — DISMISSED** [review]
LGTM, thank you @mhajas.

**Review by hmlnarik — DISMISSED** [review]

**Comment by mhajas** [comment]
@hmlnarik Have you seen the errors from GHA in other PRs? I tried locally and I am not able to reproduce any of those.

**Review by vramik — APPROVED** [review]
LGTM, thank you @mhajas

File: model/map/src/main/java/org/keycloak/models/map/realm/MapRealmEntity.java [hunk 2]
```diff
@@ -1024,7 +1025,7 @@ public boolean removeClientInitialAccess(String id) {
         return removed;
     }
 
-    public Stream<MapClientInitialAccessEntity> getClientInitialAccesses() {
-        return clientInitialAccesses.values().stream();
+    public Collection<MapClientInitialAccessEntity> getClientInitialAccesses() {
```
Comment (hmlnarik) [inline model/map/src/main/java/org/keycloak/models/map/realm/MapRealmEntity.java:1028]: Is this change necessary? The `Collection` return type is the only one in the entities so it seem inconsistent with the rest of the cide.

**Comment by hmlnarik** [comment]
> @hmlnarik Have you seen the failures from GHA in other PRs? I tried locally and I am not able to reproduce any of those.

VerifyProfileTest no, the others can be found in the JIRA.

File: model/map/src/main/java/org/keycloak/models/map/realm/MapRealmEntity.java (hunk 2 above)
Comment (mhajas) [reply model/map/src/main/java/org/keycloak/models/map/realm/MapRealmEntity.java:1028]: This is because we don't have [exists](https://github.com/keycloak/keycloak/pull/7931/files#diff-f785fd9ed631f0d3e1e9917ed2a9fdd4d900baa992038ce9afa7fdf2c7f503a4R113) and [not-exists](https://github.com/keycloak/keycloak/pull/7931/files#diff-f785fd9ed631f0d3e1e9917ed2a9fdd4d900baa992038ce9afa7fdf2c7f503a4R129) methods working with Stream, only with collections.

We have a `Collection` returned in many places across all entities, but we usually use `Set` or `List`. Do we want to refactor it and use `Stream`s everywhere?

Comment (hmlnarik) [reply model/map/src/main/java/org/keycloak/models/map/realm/MapRealmEntity.java:1028]: You're right, we can keep it changed as is here for now

**Review by hmlnarik — APPROVED** [review]
Thanks for the PR!
